hazelcast/src/main/java/com/hazelcast/map/impl/MapMigrationAwareService.java

```diff
@@ -323,6 +330,11 @@ private void depopulateIndexes(PartitionMigrationEvent event) {
 
             Indexes.markPartitionAsUnindexed(event.getPartitionId(), indexesSnapshot);
         }
+
+        ILogger logger = mapServiceContext.getNodeEngine().getLogger(getClass());
```

Comment:
Can you initialise the logger in the constructor?

Reply:
done